src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,309 @@
+        RedisSessionT8 redisSession = (RedisSessionT8)session;
+        Hashtable<String, Object> attributes = new Hashtable();
+        for (Enumeration<String> enumerator = redisSession.getAttributeNames(); enumerator.hasMoreElements();) {
+            String key = enumerator.nextElement();
+            attributes.put(key, redisSession.getAttribute(key));
+        }
+        return attributes;
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END Utils
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START REDIS
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Method to saved session in redis
+     * @param session
+     * @return
+     * @throws IOException
+     */
+    protected boolean saveSession(Session session) throws IOException {
+        boolean errorSaving = true;
+        //Get the session Id
+        String id = session.getId();
+        Hashtable<String, Object> metadata = getMetadata(session);
+        Hashtable<String, Object> attributes = getAttributes(session);
+        try{
+
+            byte[] encodedMetadata = Base64.getEncoder().encode(SerializationUtils.serialize(metadata));
+            byte[] encodedAttributes = Base64.getEncoder().encode(SerializationUtils.serialize(attributes));
+
+            withRedis(
+                    (Jedis jedis)-> {
+                        jedis.setex((id + REDIS_ATTRIBUTES_KEY).getBytes(), session.getMaxInactiveInterval(), encodedAttributes);
+                        jedis.set((id + REDIS_METADATA_KEY).getBytes(), encodedMetadata);
+                        return 0;
+                    }
+            );
+
+            errorSaving = false;
+        }catch (Exception e){
+            log.error("Error - Context: saveToRedis. Description: " + e.getMessage());
+        }
+        return errorSaving;
+    }
+
+    private interface RedisCallback<T> {
+        T apply(Jedis jedis) throws Exception;
+    }
+
+    private <T> T withRedis(RedisCallback<T> callback) throws Exception {
+        Objects.requireNonNull(redisConnectionPool, "redis connection pool must be initialized");
+        Objects.requireNonNull(callback, "callback needs to be provided");
+        Jedis jedis = null;
+        Transaction transaction = null;
+        try {
+            jedis = redisConnectionPool.getResource();
+            transaction = jedis.multi();
+            T result = callback.apply(jedis);
+            transaction.exec();
+            redisConnectionPool.returnResourceObject(jedis);
+            return result;
+        } catch (Exception e) {
+            transaction.discard();
```

Comment:
You need to only call `transaction.discard` if it is not null. This would be a `NullPointerException` if line 265 through an exception.